fishnet.py

```diff
@@ -1714,6 +1789,101 @@ def cmd_run(args):
         print("Ignored deprecated option --threads. Did you mean --cores?")
         print()
 
+    return cores, threads, instances, memory, user_backlog, system_backlog
+
+
+def cmd_benchmark(args):
+    conf = load_conf(args)
+    cores, threads, instances, memory, _, _ = display_config(args, conf)
+
+    buckets = [0] * instances
+    for i in range(0, cores):
+        buckets[i % instances] += 1
+
+    instance_memory = memory // instances
+    workers = [BenchmarkWorker(conf, bucket, instance_memory) for bucket in buckets]
+
+    logging.info("Starting %d workers" % len(workers))
+
+    # Start all threads
+    for i, worker in enumerate(workers):
+        worker.set_name("><> %d" % (i + 1))
+        worker.setDaemon(True)
+        worker.start()
+
+    # Note: Waiting with timeout on worker.running would be nicer.
+    time.sleep(3)
+    for worker in workers:
+        if not worker.running:
+            logging.warning("Worker %s never started", worker.name)
+    logging.info("All workers started")
+
+    # Let SIGTERM and SIGINT gracefully terminate the program
+    handler = SignalHandler()
+
+    finished = False
+    try:
+        # Stop workers after job finishes
+        for worker in workers:
+            worker.stop_soon()
+
+        # Wait for jobs to finish
+        for worker in workers:
+            logging.info("Waiting on worker %s" % worker.name)
+            worker.finished.wait()
```

Comment:
`cmd_run` limits this wait, in order to get to fatal errors of another worker sooner.

Reply:
It doesn't this is a direct copy of [fishnet.py#L1783-L1793](https://github.com/niklasf/fishnet/blob/master/fishnet.py#L1783-L1793), that being said I think both should be changed to have a timeout in a follow up PR.

Reply:
Ah, nevermind. I thought this is comparable to the health check [here](https://github.com/niklasf/fishnet/blob/7e115e80c0719c196b371644003cc23d2be0b85b/fishnet.py#L1748-L1753).